File: Exercise/LuoGu/Code_2023_7_23/P1200.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;

int main(void)
{

    string start_name, group_name;

    cin >> start_name >> group_name;

    long start_num = 1, group_num = 1;
    for (int i = 0; i < start_name.length(); i++)
    {
        start_num *= (start_name[i] - 'A' + 1);
    }
    for (int i = 0; i < group_name.length(); i++)
    {
        group_num *= (group_name[i] - 'A' + 1);
    }

    if (start_num % 47 == group_num % 47)
    {
        cout << "GO" << endl;
    }
    else
    {
        cout << "STAY" << endl;
    }
    return 0;
}

File: Exercise/LuoGu/Code_2023_7_23/P1200.java
package Code_2023_7_23;

import java.util.Scanner;

public class P1200 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String starName = sc.nextLine();
        String groupName = sc.nextLine();

        long starNum = 1, groupNum = 1;

        for (int i = 0; i < starName.length(); i++) {
            starNum *= (starName.charAt(i) - 'A' + 1);
        }

        for (int i = 0; i < groupName.length(); i++) {
            groupNum *= (groupName.charAt(i) - 'A' + 1);
        }

        if (starNum % 47 == groupNum % 47) {
            System.out.println("GO");
        } else {
            System.out.println("STAY");
        }

        sc.close();
    }
}
